### PR TITLE
Recently used tools

### LINKED ISSUE #12: Add functionality to display most recently used tools at the top of tool list
**Description**:
As a user 
I need a "Recently used" label feature 
So that I quickly navigate to the tool I am frequently using.

We need to implement a functionality where, when a tool is clicked or used, it is moved to the top of the tools list. Additionally, the first three most recently used tools should have a "Recently Used" label. This order and the "Recently Used" status should persist even after refreshing the page by storing the data as a JSON object token.

**Acceptance Criteria**:

- [ ] When a tool is clicked or used, it moves to the top of the tools list.
- [ ] The first three tools in the list should display a "Recently Used" label.
- [ ] The tools list should automatically reorder after a tool is used
    - [ ] The most recently used tool appearing first.
- [ ] The state should be stored as a JSON object token. 
- [ ] The recently used tools persist across page refreshes.
- [ ] The order of the tools should remain consistent during 
    - [ ] the session and 
    - [ ] after refreshing the page.
- [ ] The "Recently Used" labels should remain consistent during 
     - [ ] the session and 
     - [ ] after refreshing the page.
- [ ] This feature should be implemented for all tools on the tools page.

image

### PR DESCRIPTION
**Pull request by haneenalsewari**
Fixes #12
The "Recently Used" feature was added to the tools page. When a tool is clicked, it is moved to the top of the tools list. The first three tools in the list will display a "Recently Used" badge. This order and the recently used status will persist across page refreshes by storing the data as a JSON object in local storage.
The team needed a functionality that tracks user interactions with the tools and maintains an order for the most recently used tools. This enhances the user experience by providing quick access to frequently used tools.
The changes involved adding logic to track which tool was clicked, updating the tool order accordingly, and saving that order in localStorage. Tools are displayed in order of recency, and badges are added for the top three recently used tools. The tool list is re-rendered after every click to ensure the order is maintained. Finally, the state is persisted across page refreshes by utilizing the browser’s localStorage.

### CODE REVIEW

**Comment by kate-holdener** [comment]
@haneenalsewari I believe this PR was closed because develop branch is no longer used in this repo. Your PRs should go to the main branch. When you start working on a feature/fix, create a branch starting the latest code in main.

File: web/js/playground.js [hunk 1]
```diff
@@ -12,24 +12,54 @@ import { default as UTILS } from 'https://centerfordigitalhumanities.github.io/r
...
+    // remove the tools if it already exists in the list
+    if (existingIndex !== -1) {
+        recentTools.splice(existingIndex, 1);
+    }
+
+    // add the clicked tool to the top of the list
+    recentTools.unshift(clickedTool);
+
+    const toolIndex = ToolsCatalog.findIndex(tool => tool.label.toLowerCase() === clickedTool.label.toLowerCase());
+    // console.log('tool index in toolscatalog:', toolIndex);
```
Comment (alar12) [inline web/js/playground.js:50]: Clear the commented out code before pushing to your branch/ making a PR

File: web/js/playground.js [hunk 2]
```diff
@@ -12,24 +12,54 @@ import { default as UTILS } from 'https://centerfordigitalhumanities.github.io/r
...
+    const recentTools = getRecentlyUsedTools();
+    const existingIndex = recentTools.findIndex(tool => tool.label.toLowerCase() === clickedTool.label.toLowerCase());
+
+    console.log('clicked tool:', clickedTool);
+    console.log('existing index of tool in recent list:', existingIndex);
+    // remove the tools if it already exists in the list
+    if (existingIndex !== -1) {
+        recentTools.splice(existingIndex, 1);
+    }
+
```
Comment (alar12) [inline web/js/playground.js:45]: You have almost implemented it but the functionality blocks opening the tool. Also only navPlace Object and T-PEN are getting the label on top. We need it for all the tools in tool page.

**Review by alar12 — CHANGES_REQUESTED** [review]
Good Work!
You have almost implemented it but the functionality blocks opening a tool. Also only navPlace Object and T-PEN are getting the label on top. We need it for all the tools in tool page. Also the tools list should automatically reorder after a tool is used, with the most recently used tool appearing first.

**Review by alar12 — APPROVED** [review]
Perfect!! The functionality is working as mentioned in the requirements.

**Pull request merged**